images/php-fpm/entrypoints/50-ssmtp.sh

```diff
@@ -19,30 +19,16 @@ fi
 
 if [ ${SSMTP_MAILHUB+x} ]; then
   echo -e "\nmailhub=${SSMTP_MAILHUB}" >> /etc/ssmtp/ssmtp.conf
-else
-  # check if we find a mailhog on 172.17.0.1:1025
-  if nc -z -w 1 172.17.0.1 1025 &> /dev/null; then
-    echo -e "\nmailhub=172.17.0.1:1025" >> /etc/ssmtp/ssmtp.conf
-    return
-  fi
-  # check if mxout.lagoon.svc can do smtp TLS
-  if nc -z -w 1 mxout.lagoon.svc 465 &> /dev/null; then
-    echo -e "UseTLS=Yes\nmailhub=mxout.lagoon.svc:465" >> /etc/ssmtp/ssmtp.conf
-    return
-  fi
-  # Fallback: check if mxout.lagoon.svc can do regular 25 smtp
-  if nc -z -w 1 mxout.lagoon.svc 25 &> /dev/null; then
-    echo -e "\nmailhub=mxout.lagoon.svc:25" >> /etc/ssmtp/ssmtp.conf
-    return
-  fi
-  # check if mxout.default.svc can do smtp TLS
-  if nc -z -w 1 mxout.default.svc 465 &> /dev/null; then
-    echo -e "UseTLS=Yes\nmailhub=mxout.default.svc:465" >> /etc/ssmtp/ssmtp.conf
-    return
-  fi
-  # Fallback: check if mxout.default.svc can do regular 25 smtp
-  if nc -z -w 1 mxout.default.svc 25 &> /dev/null; then
-    echo -e "\nmailhub=mxout.default.svc:25" >> /etc/ssmtp/ssmtp.conf
-    return
-  fi
-fi
+# check if we find a mailhog on 172.17.0.1:1025
+elif nc -z -w 1 172.17.0.1 1025 &> /dev/null; then
+  echo -e "\nmailhub=172.17.0.1:1025" >> /etc/ssmtp/ssmtp.conf
+  return
+# check if mxout.lagoon.svc can do smtp TLS
+elif nc -z -w 1 mxout.lagoon.svc 465 &> /dev/null; then
+  echo -e "UseTLS=Yes\nmailhub=mxout.lagoon.svc:465" >> /etc/ssmtp/ssmtp.conf
+  return
+# Fallback: check if on Lagoon then assume mxout.lagoon.svc can do regular 25 smtp
+elif [[ ! -z ${LAGOON_PROJECT} ]]; then
```

Comment:
double `[[` are a bash way of running `if`, but we're only in `shell` here, so this probably will fail

Reply:
I tested this by running this one-liner through both `bash` and `sh`; it seems to work fine with both. :)

```if [[ ! -z ${LAGOON_PROJECT} ]]; then echo "test"; fi```

Reply:
yes, it probably works because you don't actually have `sh` installed, but your `bash` is probably used to run `sh`, see an example on MacOS:
```
❯ /bin/sh --help
GNU bash, version 3.2.57(1)-release-(x86_64-apple-darwin19)
```

`[[` is though not officially POSIX compatible so if we ever change the underlining docker images which has an POSIX only compatible shell then we got a problem, see https://stackoverflow.com/questions/669452/is-double-square-brackets-preferable-over-single-square-brackets-in-ba for more infos.

we try to keep the scripts as far as POSIX compatible as possible, so let's do it here as well :)

Reply:
Today I learned; thanks for the link! I've updated accordingly